_episodes/06-testing.md

```diff
@@ -261,43 +245,16 @@ If you're interested, you can find blog posts that compare these tools, e.g by
+
 
 > ## Hands on exercise:
 >
 >  - Create a GitHub account if you don't have one.
->  - Create a new repository and clone it, so you have a local version.
->  - Login to Travis and link your repository. 
->  - Create a text file with a simple data, you can copy our example:
-~~~
-number of dogs, number of cats, female(1)/male(0), age
-4, 0, 1, 33
-0, 4, 1, 44
-1, 1, 1, 21
-0, 0, 0, 66
-1, 0, 1, 47
-2, 1, 0, 31
-0, 2, 0, 22
-2, 0, 1, 51
-~~~
->  - Create `pets.py` file and function `mean(filename, pet_type)` that returns average age 
-of cat/dog owners.  
->  - Create a `test_pets.py` and a test function that checks proper results of 
->  the `mean` function for `pet_type=dog` and `pet_type=cat`. 
->  You can try to use [`pytest.mark.parametrize` option](http://doc.pytest.org/en/latest/parametrize.html).
->  - Run `pytest` locally and correct your `mean` function until all tests pass. 
->  - Create `requiremnts.txt` to ask for pytest library (not older than 3.0 version) 
-> and other libraries you might want to use. 
->  - Create `.travis.yml` file, ask for at least two different python version, 
-> install all libraries from `requiremnts.txt`, and run pytest to test your code. 
->  - Add everything to the repository and push to your GitHub. 
-> If Travis is set correctly, you should see that it builds various python environments 
-> and tests your code in each of them.
->  - If everything works you can come back to your local version of the repository, 
->  and think how would you like your code to behave if file doesn't contain data you expect
->  (e.g. raise exception if a file doesn't contain "number of dogs" or contain negative numbers,
->  remove a row that is incomplete)
->  - Create `test_input.py` and a few testing function that tests if your code behaves as you 
->  expect when file contain incorrect or incomplete data (you might want to create new text files 
->  with examples you want to test). Run pytest and improve your code until all test pass. 
->  - Add new tests to repository and run them using Travis.
->  - You migh want to create a Docker container and use it with the Travis platform.
\ No newline at end of file
+>  - Fork the Simple Workflow repository and clone your own fork to your computer.
+>  - Create a test directory and write a few unit tests for the function from `run_demo_workflow.py`. Run the test locally on your computer. 
+>  - Write some regression tests and compare your results to the results from `expected_output`. Run the test locally.
+>  - Login to Travis and link your repository.
+>  - Create `.travis.yml` file and run your tests using the Travis platform.
+>  - Try to use other CI platforms, e.g. CircleCI to do the same.
+>  - Try to run your tests using a Docker container (see the [README](https://github.com/ReproNim/simple_workflow#1-to-execute-demo-with-docker)) on a CI platform (note that various CI platform have various memory and time limitations).
```

Comment:
This is what we have to keep refining with this modules, to really turn them into lessons.